src/main/test/cz/muni/fi/DaoTests/ComponentDaoTest.java

```diff
@@ -0,0 +1,92 @@
+package cz.muni.fi.DaoTests;
+
+import cz.muni.fi.PersistenceApplicationContext;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.PersistenceContext;
+import javax.persistence.PersistenceUnit;
+
+/**
+ * Created by R. Tamas, 445241.
+ */
+
+@ContextConfiguration(classes = PersistenceApplicationContext.class)
+public class ComponentDaoTest extends AbstractTestNGSpringContextTests {
+
+    @PersistenceContext
+    private EntityManager em;
+
+    @PersistenceUnit
+    private EntityManagerFactory emf;
+
+    @Autowired
+    private ComponentDaoImpl componentManager = new ComponentDaoImpl();
+
+
+    private Component engine1 = new Component();
+    private Component engine2 = new Component();
+    private Component engine3 = new Component();
+
+    @BeforeClass
+    public void setup() {
+        engine1.setName("Engine 1");
+        engine1.setType(EComponentType.ENGINE);
+        engine1.setIsAvailable(true);
+
+        engine2.setName("Engine 2");
+        engine2.setType(EComponentType.ENGINE);
+        engine2.setIsAvailable(false);
+
+        engine3.setName("Engine 3");
+        engine3.setType(EComponentType.ENGINE);
+        engine3.setIsAvailable(true);
+
+        em.persist(engine1);
+        em.persist(engine2);
+        em.persist(engine3);
+    }
+
+    @Test
+    public void listAvailableComponents() {
```

Comment:
Pls name test methods in this format listAvailableComponentsTest()

Reply:
And also move the test file outside of main dir.. wrote test dir structure on messenger